Sources/Client/Session.swift

```diff
@@ -2,6 +2,98 @@ import UIKit.UIApplication
 import Instrumentation
 import Combine
 
+final class Session {
+    private var id: String
+    private var startTime: Date
+    private var backgroundTime: Date?
+    private var options: SessionOptions
+    private var cancellables = Set<AnyCancellable>()
+    var sessionAttributes: [String: String] {
+        [
+            "session.id": id,
+            "session.start_time": String(format: "%.0f", startTime.timeIntervalSince1970)
+        ]
+    }
+    private var appState = AppLifecycleState.unattached
+    
+    var sessionInfo: SessionInfo {
+        .init(
+            id: id,
+            startTime: startTime
+        )
+    }
+    
+    init(
+        id: String = UUID().uuidString,
+        startTime: Date = Date.now,
+        options: SessionOptions
+    ) {
+        self.id = id
+        self.startTime = startTime
+        self.options = options
+    }
+    
+    func start(
+        onWillEndSession: (@Sendable (_ sessionId: String) -> Void)?,
+        onDidStartSession: (@Sendable (_ sessionId: String) -> Void)?
+    ) {
+        guard let onWillEndSession, let onDidStartSession else { return }
+        
+        Publishers.MergeMany(
+            NotificationCenter.default.publisher(for: UIApplication.didEnterBackgroundNotification),
+            NotificationCenter.default.publisher(for: UIApplication.didBecomeActiveNotification),
+// Keeping this as reference
+//          NotificationCenter.default.publisher(for: UIApplication.didFinishLaunchingNotification),
+//          NotificationCenter.default.publisher(for: UIApplication.willResignActiveNotification),
+//          NotificationCenter.default.publisher(for: UIApplication.willEnterForegroundNotification),
+//          NotificationCenter.default.publisher(for: UIApplication.willTerminateNotification)
```

Comment:
will we add more notifications in the future? is that what this testing was for?